coderd/activitybump.go

```diff
@@ -0,0 +1,77 @@
+package coderd
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/coderd/database"
+)
+
+// activityBumpWorkspace automatically bumps the workspace's auto-off timer
+// if it is set to expire soon.
+func activityBumpWorkspace(log slog.Logger, db database.Store, workspace database.Workspace) {
+	// We set a short timeout so if the app is under load, these
+	// low priority operations fail first.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	err := db.InTx(func(s database.Store) error {
+		build, err := s.GetLatestWorkspaceBuildByWorkspaceID(ctx, workspace.ID)
+		log.Debug(ctx, "build", slog.F("build", build))
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil
```

Comment:
Is there a reason we don't want to surface an error? It seems like an inconsistency to me that a workspace is being bumped but doesn't exist.

Reply:
Since the bump occurs asynchronously, there is no guarantee the workspace exists when this line executes. But, that is not an error.

Reply:
That makes sense 👍🏻. Should we btw consider moving the log above to blow the error handling?